sdk/communication/AzureCommunicationChat/Source/ChatThreadClient.swift

```diff
@@ -0,0 +1,348 @@
+        chatMessageId: String,
+        withOptions options: AzureCommunicationChatService.UpdateChatMessageOptions? = nil,
+        completionHandler: @escaping HTTPResultHandler<Void>
+    ) {
+        service
+            .update(
+                chatMessage: chatMessage,
+                chatThreadId: threadId,
+                chatMessageId: chatMessageId,
+                withOptions: options
+            ) { result, httpResponse in
+                switch result {
+                case .success:
+                    completionHandler(.success(()), httpResponse)
+
+                case let .failure(error):
+                    completionHandler(.failure(error), httpResponse)
+                }
+            }
+    }
+
+    /// Deletes a message.
+    /// - Parameters:
+    ///    - messageId : The message id.
+    ///    - options: Delete chat message options
+    ///    - completionHandler: A completion handler that receives a status code on success.
+    public func delete(
+        chatMessageId: String,
+        options: AzureCommunicationChatService.DeleteChatMessageOptions? = nil,
+        completionHandler: @escaping HTTPResultHandler<Void>
+    ) {
+        service
+            .deleteChatMessage(
+                chatThreadId: threadId,
+                chatMessageId: chatMessageId,
+                withOptions: options
+            ) { result, httpResponse in
+                switch result {
+                case .success:
+                    completionHandler(.success(()), httpResponse)
+
+                case let .failure(error):
+                    completionHandler(.failure(error), httpResponse)
+                }
+            }
+    }
+
+    /// Gets a list of messages from a ChatThread.
+    /// - Parameters:
+    ///    - options: List messages options.
+    ///    - completionHandler: A completion handler that receives the list of messages on success.
+    public func listMessages(
+        withOptions options: AzureCommunicationChatService.ListChatMessagesOptions? = nil,
+        completionHandler: @escaping HTTPResultHandler<PagedCollection<ChatMessage>>
+    ) {
+        service.listChatMessages(chatThreadId: threadId, withOptions: options) { result, httpResponse in
+            switch result {
+            case let .success(messages):
+                completionHandler(.success(messages), httpResponse)
+
+            case let .failure(error):
+                completionHandler(.failure(error), httpResponse)
+            }
+        }
+    }
+
+    /// Adds thread participants to a ChatThread. If the participants already exist, no change occurs.
+    /// - Parameters:
+    ///    - request : The request with thread participants to be added to the thread.
+    ///    - options: Add chat participants options.
+    ///    - completionHandler: A completion handler that receives a status code on success.
+    public func add(
+        chatParticipants: AddChatParticipantsRequest,
```

Comment:
Consider accepting `[ChatParticipant]` here.

Reply:
I will leave as AddChatParticipantsRequest, generated code doesn't accept [ChatParticipant], so still have to construct AddChatParticipantsRequest anyways.

Reply:
Isn't that the purpose of a convenience layer?

Reply:
Yah, on second look I just changed this as well as the SendReadReceipt scenario